src/components/Home/Home.js

```diff
@@ -1,5 +1,137 @@
-import React from "react";
-
+import React, { useEffect, useState } from "react";
+import { useSelector, useDispatch } from "react-redux";
+import { getVideos } from "../../redux/reducers/video-listing/videosSlice";
+import { getCategories } from "../../redux/reducers/video-listing/categorySlice";
+import { BsThreeDotsVertical } from "react-icons/bs";
+import { MdVideoLibrary, MdThumbUp } from "react-icons/md";
 export default function Home() {
-  return <div></div>;
+  const videos = useSelector((state) => state.video.videos);
+  const categories = useSelector((state) => state.categories.categories);
+  const [showPanel, setShowPanel] = useState(false);
+  const [saveId, showSaveid] = useState(false);
+  const [filteredList, setFilteredList] = useState([]);
+
+  const dispatch = useDispatch();
+  useEffect(() => {
+    dispatch(getVideos());
+    dispatch(getCategories());
+  }, [dispatch]);
+
+  useEffect(() => {
+    setFilteredList(videos);
+  }, [videos]);
+  const handleShowPanel = (inputId) => {
+    showSaveid(inputId);
+    const videoPresent = videos.some((video) => video._id === inputId);
+
+    if (videoPresent) {
+      setShowPanel(true);
+    }
+  };
+  const sortByTag = (videos, categoryName) => {
+    let filterArr = videos.filter((video) => {
+      const category = video.categoryName.toLowerCase();
+      const selectedCategory = categoryName.toLowerCase();
+      if (category === selectedCategory) {
+        return video;
+      }
+    });
+
+    setFilteredList(filterArr);
+    if (categoryName === "all") {
+      setFilteredList(videos);
+    }
+  };
+
+  return (
+    <div className="d-flex flex-column">
+      <div className="mt-3 mb-1 gray-border-chip-container pb-2 pt-2">
+        <span
+          className="card-category-txt  white-text-color cursor-pointer"
+          onClick={() => sortByTag(videos, "all")}
+        >
+          All
+        </span>
+        {categories.map(({ categoryName }, idx) => {
+          return (
+            <span
+              className="card-category-txt  white-text-color cursor-pointer"
+              key={idx}
+              onClick={() => sortByTag(videos, categoryName)}
+            >
+              {categoryName}
+            </span>
```

Comment:
Suggestion: On hover or when the particular category is active give the indication by styling it.